test/excludeverify/ExcludeVerifyTest.gmk

```diff
@@ -0,0 +1,149 @@
+#
+# Copyright (c) 2012, 2022, Oracle and/or its affiliates. All rights reserved.
+# DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+#
+# This code is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License version 2 only, as
+# published by the Free Software Foundation.  Oracle designates this
+# particular file as subject to the "Classpath" exception as provided
+# by Oracle in the LICENSE file that accompanied this code.
+#
+# This code is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+# version 2 for more details (a copy is included in the LICENSE file that
+# accompanied this code).
+#
+# You should have received a copy of the GNU General Public License version
+# 2 along with this work; if not, write to the Free Software Foundation,
+# Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+# or visit www.oracle.com if you need additional information or have any
+# questions.
+#
+
+#----------------------------------------------------------------------
+
+$(BUILDTESTDIR)/excludeverify.format.1.ok: \
+                $(JTREG_IMAGEDIR)/lib/jtreg.jar \
+		$(JTREG_IMAGEDIR)/lib/javatest.jar
+	$(RM) $(@:%.ok=%/work) $(@:%.ok=%/report)
+	$(MKDIR) -p $(@:%.ok=%)
+	$(JDKHOME)/bin/java \
+		-Djavatest.regtest.showCmd=true \
+		-jar $(JTREG_IMAGEJARDIR)/jtreg.jar \
+		-w:$(@:%.ok=%/work) \
+		-r:$(@:%.ok=%/report) \
+		-exclude:$(TESTDIR)/excludeverify/ProblemList/format.1.txt \
+		-verifyexclude \
+		$(TESTDIR)/excludeverify \
+		> $(@:%.ok=%/log 2>&1) \
+	    && (cat $(@:%.ok=%/log) ; exit 1) \
+		|| (exit 0)
+	$(GREP) -s "Must follow:" $(@:%.ok=%/log 2>&1)
+	$(GREP) -s "The fully qualified test must exists." $(@:%.ok=%/log 2>&1) && exit 1 || exit 0
```

Comment:
Here and on similar lines, the appearance of `2>&1` _inside_ the `$(@...)` expression seems weird to the point of wrong.  I don't know what you're trying to do of if any redirection is actually necessary here. If you copied this code from elsewhere, I'd be interested to know where from.

Reply:
I'm pretty sure I copied from https://github.com/openjdk/jtreg/blob/master/test/exclude/ExcludeTest.gmk, the same pattern is found on https://github.com/openjdk/jtreg/blob/master/test/exclude/ExcludeTest.gmk#L44.

Reply:
I am still not sure whether there's something wrong with these or not, but if there is, that other test should also be looked at.

Reply:
The code might work "by accident", but is for sure not how it is meant to be.

There are two parts to this expression:
1. `$(@:%.ok=%/log)`. This is an expression to generate a string from `$@` by replacing the trailing `.ok` with `/log` -- in other words, based on the name of the target `$@` generate a related name for the log file
2. `2>&1`. This is plain old shell redirection, and should be directly on the `grep` command line, and not hidden in. the `$(@:...)` expression.

Reply:
I've created https://bugs.openjdk.org/browse/CODETOOLS-7903699 to track the other occurrences.